File: src/main/java/jk/kamoru/flayground/flay/domain/info/Video.java
package jk.kamoru.flayground.flay.domain.info;

import java.util.Date;
import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Data
public class Video implements Info<String> {

	String opus;
	Integer play;
	Integer rank;
	String comment;
	Date lastAccess;
	List<Tag> tags;

	public Video(String key) {
		setKey(key);
		this.play = 0;
		this.rank = 0;
		this.comment = "";
		this.lastAccess = new Date(0);
	}

	@Override
	public String getKey() {
		return opus;
	}

	@Override
	public void setKey(String key) {
		this.opus = key;
	}

}

File: server_node/src/domain/video.ts
import { Tag } from './tag';

/** Video 도메인 (Info<String> 대응) */
export interface Video {
  opus: string;
  play: number;
  rank: number;
  /** 마지막 재생 시각 (epoch ms) */
  lastPlay: number;
  /** 마지막 접근 시각 (epoch ms) */
  lastAccess: number;
  lastModified: number;
  comment: string;
  title: string;
  desc: string;
  tags: Tag[];
  /** ISO date 문자열 배열 (Java List<Date> -> JSON ISO string[]) */
  likes: string[];
}

/**
 * 새 Video 생성
 * @param opus 비디오 식별자
 */
export function createVideo(opus: string): Video {
  return {
    opus,
    play: 0,
    rank: 0,
    lastPlay: -1,
    lastAccess: -1,
    lastModified: -1,
    comment: '',
    title: '',
    desc: '',
    tags: [],
    likes: [],
  };
}
